packages/tizen_audio_manager/tizen/src/audio_manager.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_PLUGIN_AUDIO_MANAGER_H_
+#define FLUTTER_PLUGIN_AUDIO_MANAGER_H_
+
+#include <sound_manager.h>
+
+#include <functional>
+#include <string>
+
+using VolumeChangedListener =
+    std::function<void(std::string type_str, int32_t volume)>;
+
+class AudioManager {
+ public:
+  AudioManager() {}
+  ~AudioManager() {}
+
+  std::string getCurrentPlaybackType();
+
+  int32_t getMaxLevel(std::string getMaxLevel);
+
+  void SetLevel(std::string type, int32_t volume);
+
+  int32_t GetLevel(std::string type);
+
+  static void OnVolumeChanged(sound_type_e type, unsigned int volume,
+                              void *user_data);
+
+  void AddVolumeChangedCallback(VolumeChangedListener volume_changed_listener);
+
+  void RemoveVolumeChangedCallback();
```

Comment:
The name `Add...` implies that the method can be called multiple times but in reality it will be a memory leak. You might rename and merge these to a single method `SetVolumeChangedHandler` to avoid any misunderstanding. You can also remove the extra `registered_volume_changed_cb_` flag by moving the implementation of the methods to the constructor and destructor of the class if you want (please refer to similar patterns used by other plugins such as [tizen_app_manager.h](https://github.com/flutter-tizen/plugins/blob/master/packages/tizen_app_manager/tizen/src/tizen_app_manager.h)).

Reply:
> You can also remove the extra registered_volume_changed_cb_ flag by moving the implementation of the methods to the constructor and destructor of the class if you want (please refer to similar patterns used by other plugins such as [tizen_app_manager.h](https://github.com/flutter-tizen/plugins/blob/master/packages/tizen_app_manager/tizen/src/tizen_app_manager.h)).

I removed registered_volume_changed_cb_ and changed method name(Register/Unregister).

Reply:
I didnt move Register/Unregister method to the constructor/destructor.
Because, in the `TizenAppManager` structure, when the tizen api inside the constructor fails, it seems that the `ret`'s error is not handled. I thought it would be better to keep this structure rather than adding separate code for error handle.
(And we should check the error handling of `app_manager_set_app_context_event_cb` of `TizenAppManager`.)

Reply:
Note that in that case `TizenAudioManager` doesn't have to be a singleton class (`TizenAudioManager::GetInstance`).

Reply:
I removed singleton code